[Feat/Chore] 개발환경 구성 및 ERD 엔티티 구현

happymink opened the pull request:
## 변경사항

docker, yml 등 개발 환경에 필요한 설정 및 ERD 엔티티를 작성했습니다.

### Issue Link - #6 

## 체크리스트

- [x] 개발환경이 컨테이너를 통해 적절히 구성이 되었는가
- [x] Local 환경에서 DB 접속이 원활하게 진행되었는가
- [x] ERD 구성이 적절했는가

## 참고
### ERD
![tripvote2](https://github.com/Strong-Potato/TripVote-BE/assets/26517061/ad8062e8-ada6-4bde-91ed-c1722a09a78f)

happymink committed:
[Chore] 데이터베이스 설정

`+` Dockerfile, init.sql, docker-compose
`+` 상황별 application.yml
`-` application.properties

happymink committed:
[Feat] Entity 개발

`+` domain 패키지 엔티티
`+` global 패키지 엔티티

laigasus commented on src/main/java/fc/be/tripvote/global/util/ContentTypeIdConverter.java:
```diff
@@ -0,0 +1,30 @@
+                return contentTypeId;
+            }
+        }
+        throw new IllegalArgumentException("Unknown database value: " + dbData);
```
추후 예외처리에서 공통, Biz 별 예외처리가 필요하겠습니다

NoJaeHyuk commented on src/main/java/fc/be/tripvote/domain/space/vo/VoteStatus.java:
```diff
@@ -0,0 +1,13 @@
+    VOTING("투표중"),
+    DONE("투표완료");
+
+    private final String text;
```
상태값에 대한 설명이면 description 표현이 어떤가요?

laigasus reviewed (approved):
인프라(ex. Docker) 환경 설정 이상 없습니다 👍

Dr-KoKo commented on src/main/java/fc/be/tripvote/domain/space/vo/VoteStatus.java:
```diff
@@ -0,0 +1,13 @@
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
```
ContentTypeId처럼 @AllArgsConstructor로 변경하는 것이 이해하기 더 좋은 것 같아요!

Dr-KoKo reviewed (commented):
생성자 통일하면 좋을 것 같습니다.

happymink committed:
[Refactor] VoteStatus 수정

`#` RequiredArgsConstructor -> AllArgsConstructor 수정
`#` field Naming, text -> description 수정

Dr-KoKo reviewed (approved):
훨씬 보기 좋네요!
